n3fit/src/n3fit/hyper_optimization/mongofiletrials.py

```diff
@@ -0,0 +1,290 @@
+
+    def start_mongo_workers(
+        self, workdir=None, exp_key=None, poll_interval=0.1, use_subprocesses=False
+    ):
+        """Initiates all mongo workers simultaneously."""
+        # get the number of gpu cards, if any
+        gpus_all_physical_list = get_physical_gpus()
+        num_gpus_available = len(gpus_all_physical_list)
+        if not num_gpus_available:
+            log.warning("No GPUs found in the system.")
+
+        # launch mongo workers
+        for i in range(self.num_workers):
+            # construct the command to start a hyperopt-mongo-worker
+            args = [
+                "hyperopt-mongo-worker",
+                "--mongo",
+                f"{self.db_host}:{self.db_port}/{self.db_name}",
+            ]
+            if workdir:
+                args.extend(["--workdir", workdir])
+            if exp_key:
+                args.extend(["--exp-key", exp_key])
+            args.extend(["--poll-interval", str(poll_interval)])
+            if use_subprocesses:
+                args.append("--no-subprocesses")
+
+            # start the worker as a subprocess
+            try:
+                my_env = os.environ.copy()
+
+                if num_gpus_available:
+                    # set CUDA_VISIBLE_DEVICES environment variable
+                    # the GPU index assigned to each worker i is given by mod(i, num_gpus_available)
+                    my_env["CUDA_VISIBLE_DEVICES"] = str(i % num_gpus_available)
+                    # set tensorflow memory growth
+                    my_env["TF_FORCE_GPU_ALLOW_GROWTH"] = "true"
+                    # avoid memory fragmentation issues?
+                    # my_env["TF_GPU_ALLOCATOR"] = "cuda_malloc_async"
+
+                # run mongo workers
+                # we could use stdout=subprocess.DEVNULL and stderr=subprocess.DEVNULL in Popen to suppress output info
+                worker = subprocess.Popen(args, env=my_env)
+                self.workers.append(worker)
+                log.info(f"Started mongo worker {i+1}/{self.num_workers}")
+            except OSError as err:
+                msg = f"Failed to execute {args}. Make sure you have MongoDB installed."
+                raise EnvironmentError(msg) from err
+
+    def stop_mongo_workers(self):
+        """Terminates all active mongo workers."""
+        for worker in self.workers:
+            try:
+                worker.terminate()
+                worker.wait()
+                log.info(f"Stopped mongo worker {self.workers.index(worker)+1}/{self.num_workers}")
+            except Exception as err:
+                log.error(
+                    f"Failed to stop mongo worker {self.workers.index(worker)+1}/{self.num_workers}: {err}"
+                )
+
+    def compress_mongodb_database(self):
+        """Saves MongoDB database as tar file"""
+        # check if the database exist
+        if not os.path.exists(f"{self.db_name}" and not glob.glob('65*')):
```

Comment:
Out of curiosity, why does it have to start with `65`?

Reply:
Thanks for pointing this out. After research for more info I noticed that the names of these extra directories might change depending on the system settings. To avoid this problem I added an extra options to `mongod` `--directoryperdb` which puts everything we need inside `hyperopt-db`. So, we do not need these `65` directories anymore to restart.